public/scripts/tagsList.js

```diff
@@ -1,21 +1,148 @@
-function getTagObject(content){
+let labelArray = null
+let currentFocus = -1
+
+function findLabelSuggestion(input) {
+    if (input === '')
+        return []
+    let suggestion = []
+
+    suggestion = labelArray.filter(element => {
+        if (input.length > element.length)
+            return false
+        return input.toUpperCase() === element.substring(0, input.length).toUpperCase()
+    })
+
+    return suggestion
+}
+
+function createTagObject(content) {
     let template = document.createElement('template')
-    template.innerHTML = 
-    `<div class="tag-element">
+    template.innerHTML =
+        `<div class="tag-element">
         <span class="tag-element-text" data-value=${content}>${content}</span> 
-        <button class="delete-tag-button"></button> 
+        <button class="delete-tag-button" type="button">x</button> 
     </div>`
+    template.content.querySelector('button').onclick = (event) => {
+        event.target.parentNode.remove()
+    }
     return template.content  // this is so much simple :O
+}
+
+function closeRecommendationList() {
```

Comment:
Again, back on the `autocomplete` wagon. Perhaps `closeAutocompleteList` instead of `closeRecommendationList`? Not 100% I'm on the right track, though. So feel free to correct me.